service/src/common/indexer_management_client/resolver.rs

```diff
@@ -0,0 +1,343 @@
+// Copyright 2023-, GraphOps and Semiotic Labs.
+// SPDX-License-Identifier: Apache-2.0
+
+use sqlx::PgPool;
+
+use super::schema::CostModel;
+use crate::common::{
+    indexer_error::{IndexerError, IndexerErrorCause, IndexerErrorCode},
+    types::SubgraphDeploymentID,
+};
+
+/// Query postgres indexer management server's cost models
+/// Filter on deployments if it is not empty, otherwise return all cost models
```

Comment:
Needs a bit more explanation about what happens wrt. the global model and empty vs. non-empty `deployments`